### ISSUE TITLE
Orca Learn Sentiment Example with Bigdl Backend Bug

### ISSUE THREAD

**Issue by Elena-Qiu** [intel-analytics/analytics-zoo#3920]
When running the orca learn sentiment example using bigdl backend, will get the following error:

`  File "/home/usr/downloads/spark-2.4.3-bin-hadoop2.7/python/lib/py4j-0.10.7-src.zip/py4j/java_gateway.py", line 1257, in __call__
  File "/home/usr/downloads/spark-2.4.3-bin-hadoop2.7/python/lib/py4j-0.10.7-src.zip/py4j/protocol.py", line 332, in get_return_value
py4j.protocol.Py4JError: An error occurred while calling o41.createTorchModel. Trace:
java.lang.NegativeArraySizeException
	at py4j.Base64.decode(Base64.java:321)
	at py4j.Protocol.getBytes(Protocol.java:175)
	at py4j.Protocol.getObject(Protocol.java:296)
	at py4j.commands.AbstractCommand.getArguments(AbstractCommand.java:82)
	at py4j.commands.CallCommand.execute(CallCommand.java:77)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)`

The bigdl backend is implemented as follows:

`train_iter = train_loader_creator({}, batch_size)
test_iter = test_loader_creator({}, batch_size)

if args.backend == "bigdl":
    net = model_creator({})
    optimizer = optim_creator(model=net,config={})
    orca_estimator = Estimator.from_torch(model=net,
                                          optimizer=optimizer,
                                          loss=criterion,
                                          metrics=[Accuracy()],
                                          backend="bigdl")

    orca_estimator.fit(data=train_iter, epochs=2, validation_data=test_iter,
                       checkpoint_trigger=EveryEpoch())

    res = orca_estimator.evaluate(data=test_iter)
    print("Accuracy of the network on the test images: %s" % res)
`

**Comment by hkvision** [comment]
The code is here: https://github.com/intel-analytics/analytics-zoo/pull/3918
@qiuxin2012 Do we support iterator as data?

**Comment by qiuxin2012** [comment]
Why not dataloader?
Iterator is not supported.

**Comment by Elena-Qiu** [comment]
I have changed the iterator to dataloader and it runs successfully with two workers using torch_distributed backend. But the bigdl backend still gets the same error as above.

**Comment by hkvision** [comment]
@qiuxin2012 Take a look.

**Comment by qiuxin2012** [comment]
Use something like train_loader_creator, but not train_iter.
train_iter is too big be pickled.

**Comment by Elena-Qiu** [comment]
I tried with the train_loader_creator and test_loader_creator but still got the same error.

**Comment by qiuxin2012** [comment]
Looks your model is too big to be pickled, please use a model creator function instead of model instance.

**Comment by qiuxin2012** [comment]
https://github.com/intel-analytics/analytics-zoo/blob/ee24ffcc17458490da1a42bc6ad6e5f881d41106/pyzoo/zoo/orca/learn/pytorch/estimator.py#L275
The estimator shouldn't create model the model instance here, we should pickle a model creator function to bytes and pass it to executor.(pickle only support 155MB data)

**Comment by hkvision** [comment]
Next step:
- Fix the create instance model bug mentioned above. After the fix, test with this example to see if config is too large as well.
- Do some checking and throw some clearer error message if possible.
- Any improvement to support larger size?

**Comment by hkvision** [comment]
Any update on this issue and fix? @qiuxin2012 @Le-Zheng

**Comment by Elena-Qiu** [comment]
I tried with the latest estimator.py but I still got the same error as before.

**Comment by hkvision** [comment]
Take a look @qiuxin2012 @Le-Zheng
